[Feature][Sort][Manager] Support InLongMsg in KafkaConnector

### Description


#### 问题1: 分隔符前端与后台不一致
<img width="995" alt="image" src="https://user-images.githubusercontent.com/5709212/216524453-fadb9d88-e5b7-4d30-82c5-595b554be03e.png">

<img width="542" alt="image" src="https://user-images.githubusercontent.com/5709212/216528062-e785606c-28db-493b-9486-790f244886a7.png">

![image](https://user-images.githubusercontent.com/5709212/216525641-cb150570-d4d6-4910-bec0-1ddd0b7969e7.png)


#### 问题2: csv.field-delimiter为null，直接传递到了Connector

```sql
CREATE TABLE `table_hudi-stream-v1`(
    `f1` INT,
    `f2` STRING,
    `f3` DOUBLE,
    `f4` BIGINT)
    WITH (
    'inlong.metric.labels' = 'groupId=first_stream_group_20230202&streamId=hudi-stream-v1&nodeId=hudi-stream-v1',
    'metrics.audit.proxy.hosts' = '127.0.0.1:10081',
    'topic' = 'hudi-stream-v1',
    'properties.bootstrap.servers' = '127.0.0.1:9092',
    'connector' = 'kafka-inlong',
    'scan.startup.mode' = 'earliest-offset',
    'inlong-msg.csv.array-element-delimiter' = ';',
    'inlong-msg.csv.disable-quote-character' = 'true',
    'inlong-msg.inner.format' = 'csv',
    'inlong-msg.csv.allow-comments' = 'false',
    'format' = 'inlong-msg',
    'inlong-msg.csv.field-delimiter' = 'null',
    'inlong-msg.csv.ignore-parse-errors' = 'false'
) 
```

### 问题3: 未适配InLongMsg格式


```java
Unable to create a source for reading table 'default_catalog.default_database.table_hudi-stream-v1'.

Table options are:

'connector'='kafka-inlong'
'format'='inlong-msg'
'inlong-msg.csv.allow-comments'='false'
'inlong-msg.csv.array-element-delimiter'=';'
'inlong-msg.csv.disable-quote-character'='true'
'inlong-msg.csv.field-delimiter'='null'
'inlong-msg.csv.ignore-parse-errors'='false'
'inlong-msg.inner.format'='csv'
'inlong.metric.labels'='groupId=first_stream_group_20230202&streamId=hudi-stream-v1&nodeId=hudi-stream-v1'
'metrics.audit.proxy.hosts'='127.0.0.1:10081'
'properties.bootstrap.servers'='127.0.0.1:9092'
'scan.startup.mode'='earliest-offset'
'topic'='hudi-stream-v1'
	at org.apache.flink.client.program.PackagedProgram.callMainMethod(PackagedProgram.java:372) ~[flink-clients_2.11-1.13.5.jar:1.13.5]
	at org.apache.flink.client.program.PackagedProgram.invokeInteractiveModeForExecution(PackagedProgram.java:222) ~[flink-clients_2.11-1.13.5.jar:1.13.5]
	at org.apache.flink.client.program.PackagedProgramUtils.getPipelineFromProgram(PackagedProgramUtils.java:158) ~[flink-clients_2.11-1.13.5.jar:1.13.5]
	at org.apache.flink.client.program.PackagedProgramUtils.createJobGraph(PackagedProgramUtils.java:82) ~[flink-clients_2.11-1.13.5.jar:1.13.5]
	at org.apache.flink.client.program.PackagedProgramUtils.createJobGraph(PackagedProgramUtils.java:117) ~[flink-clients_2.11-1.13.5.jar:1.13.5]
	at org.apache.inlong.manager.plugin.flink.FlinkService.submitJobBySavepoint(FlinkService.java:215) ~[classes/:?]
	at org.apache.inlong.manager.plugin.flink.FlinkService.submit(FlinkService.java:167) ~[classes/:?]
	at org.apache.inlong.manager.plugin.flink.IntegrationTaskRunner.run(IntegrationTaskRunner.java:57) ~[classes/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_292]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[?:1.8.0_292]
	at java.lang.Thread.run(Thread.java:748) ~[?:1.8.0_292]
Caused by: org.apache.flink.table.api.ValidationException: Unable to create a source for reading table 'default_catalog.default_database.table_hudi-stream-v1'.

Table options are:

'connector'='kafka-inlong'
'format'='inlong-msg'
'inlong-msg.csv.allow-comments'='false'
'inlong-msg.csv.array-element-delimiter'=';'
'inlong-msg.csv.disable-quote-character'='true'
'inlong-msg.csv.field-delimiter'='null'
'inlong-msg.csv.ignore-parse-errors'='false'
'inlong-msg.inner.format'='csv'
'inlong.metric.labels'='groupId=first_stream_group_20230202&streamId=hudi-stream-v1&nodeId=hudi-stream-v1'
'metrics.audit.proxy.hosts'='127.0.0.1:10081'
'properties.bootstrap.servers'='127.0.0.1:9092'
'scan.startup.mode'='earliest-offset'
'topic'='hudi-stream-v1'
	at org.apache.flink.table.factories.FactoryUtil.createTableSource(FactoryUtil.java:137) ~[flink-table-common-1.13.5.jar:1.13.5]
	at org.apache.flink.table.planner.plan.schema.CatalogSourceTable.createDynamicTableSource(CatalogSourceTable.java:116) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.flink.table.planner.plan.schema.CatalogSourceTable.toRel(CatalogSourceTable.java:82) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.calcite.sql2rel.SqlToRelConverter.toRel(SqlToRelConverter.java:3585) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.calcite.sql2rel.SqlToRelConverter.convertIdentifier(SqlToRelConverter.java:2507) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.calcite.sql2rel.SqlToRelConverter.convertFrom(SqlToRelConverter.java:2144) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.calcite.sql2rel.SqlToRelConverter.convertFrom(SqlToRelConverter.java:2093) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.calcite.sql2rel.SqlToRelConverter.convertFrom(SqlToRelConverter.java:2050) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.calcite.sql2rel.SqlToRelConverter.convertSelectImpl(SqlToRelConverter.java:663) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.calcite.sql2rel.SqlToRelConverter.convertSelect(SqlToRelConverter.java:644) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.calcite.sql2rel.SqlToRelConverter.convertQueryRecursive(SqlToRelConverter.java:3438) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.calcite.sql2rel.SqlToRelConverter.convertQuery(SqlToRelConverter.java:570) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.flink.table.planner.calcite.FlinkPlannerImpl.org$apache$flink$table$planner$calcite$FlinkPlannerImpl$$rel(FlinkPlannerImpl.scala:169) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.flink.table.planner.calcite.FlinkPlannerImpl.rel(FlinkPlannerImpl.scala:161) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.flink.table.planner.operations.SqlToOperationConverter.toQueryOperation(SqlToOperationConverter.java:989) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.flink.table.planner.operations.SqlToOperationConverter.convertSqlQuery(SqlToOperationConverter.java:958) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.flink.table.planner.operations.SqlToOperationConverter.convert(SqlToOperationConverter.java:283) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.flink.table.planner.operations.SqlToOperationConverter.convertSqlInsert(SqlToOperationConverter.java:603) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.flink.table.planner.operations.SqlToOperationConverter.convert(SqlToOperationConverter.java:272) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.flink.table.planner.delegation.ParserImpl.parse(ParserImpl.java:101) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.flink.table.api.internal.StatementSetImpl.addInsertSql(StatementSetImpl.java:51) ~[flink-table-api-java-1.13.5.jar:1.13.5]
	at org.apache.inlong.sort.parser.result.FlinkSqlParseResult.executeLoadSqls(FlinkSqlParseResult.java:80) ~[?:?]
	at org.apache.inlong.sort.parser.result.FlinkSqlParseResult.execute(FlinkSqlParseResult.java:61) ~[?:?]
	at org.apache.inlong.sort.Entrance.main(Entrance.java:72) ~[?:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_292]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_292]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_292]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_292]
	at org.apache.flink.client.program.PackagedProgram.callMainMethod(PackagedProgram.java:355) ~[flink-clients_2.11-1.13.5.jar:1.13.5]
	... 12 more
Caused by: org.apache.flink.table.api.ValidationException: Error creating scan format 'inlong-msg' in option space 'inlong-msg.'.
	at org.apache.flink.table.factories.FactoryUtil$TableFactoryHelper.lambda$discoverOptionalDecodingFormat$1(FactoryUtil.java:658) ~[flink-table-common-1.13.5.jar:1.13.5]
	at java.util.Optional.map(Optional.java:215) ~[?:1.8.0_292]
	at org.apache.flink.table.factories.FactoryUtil$TableFactoryHelper.discoverOptionalDecodingFormat(FactoryUtil.java:650) ~[flink-table-common-1.13.5.jar:1.13.5]
	at org.apache.inlong.sort.kafka.table.KafkaDynamicTableFactory.getValueDecodingFormat(KafkaDynamicTableFactory.java:172) ~[?:?]
	at org.apache.inlong.sort.kafka.table.KafkaDynamicTableFactory.createDynamicTableSource(KafkaDynamicTableFactory.java:327) ~[?:?]
	at org.apache.flink.table.factories.FactoryUtil.createTableSource(FactoryUtil.java:134) ~[flink-table-common-1.13.5.jar:1.13.5]
	at org.apache.flink.table.planner.plan.schema.CatalogSourceTable.createDynamicTableSource(CatalogSourceTable.java:116) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.flink.table.planner.plan.schema.CatalogSourceTable.toRel(CatalogSourceTable.java:82) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.calcite.sql2rel.SqlToRelConverter.toRel(SqlToRelConverter.java:3585) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.calcite.sql2rel.SqlToRelConverter.convertIdentifier(SqlToRelConverter.java:2507) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.calcite.sql2rel.SqlToRelConverter.convertFrom(SqlToRelConverter.java:2144) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.calcite.sql2rel.SqlToRelConverter.convertFrom(SqlToRelConverter.java:2093) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.calcite.sql2rel.SqlToRelConverter.convertFrom(SqlToRelConverter.java:2050) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.calcite.sql2rel.SqlToRelConverter.convertSelectImpl(SqlToRelConverter.java:663) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.calcite.sql2rel.SqlToRelConverter.convertSelect(SqlToRelConverter.java:644) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.calcite.sql2rel.SqlToRelConverter.convertQueryRecursive(SqlToRelConverter.java:3438) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.calcite.sql2rel.SqlToRelConverter.convertQuery(SqlToRelConverter.java:570) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.flink.table.planner.calcite.FlinkPlannerImpl.org$apache$flink$table$planner$calcite$FlinkPlannerImpl$$rel(FlinkPlannerImpl.scala:169) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.flink.table.planner.calcite.FlinkPlannerImpl.rel(FlinkPlannerImpl.scala:161) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.flink.table.planner.operations.SqlToOperationConverter.toQueryOperation(SqlToOperationConverter.java:989) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.flink.table.planner.operations.SqlToOperationConverter.convertSqlQuery(SqlToOperationConverter.java:958) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.flink.table.planner.operations.SqlToOperationConverter.convert(SqlToOperationConverter.java:283) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.flink.table.planner.operations.SqlToOperationConverter.convertSqlInsert(SqlToOperationConverter.java:603) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.flink.table.planner.operations.SqlToOperationConverter.convert(SqlToOperationConverter.java:272) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.flink.table.planner.delegation.ParserImpl.parse(ParserImpl.java:101) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.flink.table.api.internal.StatementSetImpl.addInsertSql(StatementSetImpl.java:51) ~[flink-table-api-java-1.13.5.jar:1.13.5]
	at org.apache.inlong.sort.parser.result.FlinkSqlParseResult.executeLoadSqls(FlinkSqlParseResult.java:80) ~[?:?]
	at org.apache.inlong.sort.parser.result.FlinkSqlParseResult.execute(FlinkSqlParseResult.java:61) ~[?:?]
	at org.apache.inlong.sort.Entrance.main(Entrance.java:72) ~[?:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_292]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_292]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_292]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_292]
	at org.apache.flink.client.program.PackagedProgram.callMainMethod(PackagedProgram.java:355) ~[flink-clients_2.11-1.13.5.jar:1.13.5]
	... 12 more
Caused by: org.apache.flink.table.api.ValidationException: Option 'csv.field-delimiter' must be a string with single character, but was: null
	at org.apache.flink.formats.csv.CsvFormatFactory.validateCharacterVal(CsvFormatFactory.java:178) ~[?:?]
	at org.apache.flink.formats.csv.CsvFormatFactory.validateFormatOptions(CsvFormatFactory.java:150) ~[?:?]
	at org.apache.flink.formats.csv.CsvFormatFactory.createDecodingFormat(CsvFormatFactory.java:68) ~[?:?]
	at org.apache.inlong.sort.formats.inlongmsg.InLongMsgFormatFactory.createDecodingFormat(InLongMsgFormatFactory.java:66) ~[?:?]
	at org.apache.flink.table.factories.FactoryUtil$TableFactoryHelper.lambda$discoverOptionalDecodingFormat$1(FactoryUtil.java:654) ~[flink-table-common-1.13.5.jar:1.13.5]
	at java.util.Optional.map(Optional.java:215) ~[?:1.8.0_292]
	at org.apache.flink.table.factories.FactoryUtil$TableFactoryHelper.discoverOptionalDecodingFormat(FactoryUtil.java:650) ~[flink-table-common-1.13.5.jar:1.13.5]
	at org.apache.inlong.sort.kafka.table.KafkaDynamicTableFactory.getValueDecodingFormat(KafkaDynamicTableFactory.java:172) ~[?:?]
	at org.apache.inlong.sort.kafka.table.KafkaDynamicTableFactory.createDynamicTableSource(KafkaDynamicTableFactory.java:327) ~[?:?]
	at org.apache.flink.table.factories.FactoryUtil.createTableSource(FactoryUtil.java:134) ~[flink-table-common-1.13.5.jar:1.13.5]
	at org.apache.flink.table.planner.plan.schema.CatalogSourceTable.createDynamicTableSource(CatalogSourceTable.java:116) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.flink.table.planner.plan.schema.CatalogSourceTable.toRel(CatalogSourceTable.java:82) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.calcite.sql2rel.SqlToRelConverter.toRel(SqlToRelConverter.java:3585) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.calcite.sql2rel.SqlToRelConverter.convertIdentifier(SqlToRelConverter.java:2507) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.calcite.sql2rel.SqlToRelConverter.convertFrom(SqlToRelConverter.java:2144) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.calcite.sql2rel.SqlToRelConverter.convertFrom(SqlToRelConverter.java:2093) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.calcite.sql2rel.SqlToRelConverter.convertFrom(SqlToRelConverter.java:2050) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.calcite.sql2rel.SqlToRelConverter.convertSelectImpl(SqlToRelConverter.java:663) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.calcite.sql2rel.SqlToRelConverter.convertSelect(SqlToRelConverter.java:644) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.calcite.sql2rel.SqlToRelConverter.convertQueryRecursive(SqlToRelConverter.java:3438) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.calcite.sql2rel.SqlToRelConverter.convertQuery(SqlToRelConverter.java:570) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.flink.table.planner.calcite.FlinkPlannerImpl.org$apache$flink$table$planner$calcite$FlinkPlannerImpl$$rel(FlinkPlannerImpl.scala:169) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.flink.table.planner.calcite.FlinkPlannerImpl.rel(FlinkPlannerImpl.scala:161) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.flink.table.planner.operations.SqlToOperationConverter.toQueryOperation(SqlToOperationConverter.java:989) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.flink.table.planner.operations.SqlToOperationConverter.convertSqlQuery(SqlToOperationConverter.java:958) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.flink.table.planner.operations.SqlToOperationConverter.convert(SqlToOperationConverter.java:283) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.flink.table.planner.operations.SqlToOperationConverter.convertSqlInsert(SqlToOperationConverter.java:603) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.flink.table.planner.operations.SqlToOperationConverter.convert(SqlToOperationConverter.java:272) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.flink.table.planner.delegation.ParserImpl.parse(ParserImpl.java:101) ~[flink-table-planner-blink_2.11-1.13.5.jar:1.13.5]
	at org.apache.flink.table.api.internal.StatementSetImpl.addInsertSql(StatementSetImpl.java:51) ~[flink-table-api-java-1.13.5.jar:1.13.5]
	at org.apache.inlong.sort.parser.result.FlinkSqlParseResult.executeLoadSqls(FlinkSqlParseResult.java:80) ~[?:?]
	at org.apache.inlong.sort.parser.result.FlinkSqlParseResult.execute(FlinkSqlParseResult.java:61) ~[?:?]
	at org.apache.inlong.sort.Entrance.main(Entrance.java:72) ~[?:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_292]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_292]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_292]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_292]
	at org.apache.flink.client.program.PackagedProgram.callMainMethod(PackagedProgram.java:355) ~[flink-clients_2.11-1.13.5.jar:1.13.5]
	... 12 more
```



### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/KafkaExtractNode.java
+++ b/inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/KafkaExtractNode.java
@@ -18,8 +18,6 @@
 package org.apache.inlong.sort.protocol.node.extract;
 
 import com.google.common.base.Preconditions;
-import java.util.HashMap;
-import java.util.Map.Entry;
 import lombok.Data;
 import lombok.EqualsAndHashCode;
 import org.apache.commons.lang3.StringUtils;
@@ -35,14 +33,7 @@ import org.apache.inlong.sort.protocol.Metadata;
 import org.apache.inlong.sort.protocol.constant.KafkaConstant;
 import org.apache.inlong.sort.protocol.enums.KafkaScanStartupMode;
 import org.apache.inlong.sort.protocol.node.ExtractNode;
-import org.apache.inlong.sort.protocol.node.format.AvroFormat;
-import org.apache.inlong.sort.protocol.node.format.CanalJsonFormat;
-import org.apache.inlong.sort.protocol.node.format.CsvFormat;
-import org.apache.inlong.sort.protocol.node.format.DebeziumJsonFormat;
 import org.apache.inlong.sort.protocol.node.format.Format;
-import org.apache.inlong.sort.protocol.node.format.InLongMsgFormat;
-import org.apache.inlong.sort.protocol.node.format.JsonFormat;
-import org.apache.inlong.sort.protocol.node.format.RawFormat;
 import org.apache.inlong.sort.protocol.transformation.WatermarkField;
 
 import javax.annotation.Nonnull;
@@ -155,63 +146,22 @@ public class KafkaExtractNode extends ExtractNode implements InlongMetric, Metad
         Map<String, String> options = super.tableOptions();
         options.put(KafkaConstant.TOPIC, topic);
         options.put(KafkaConstant.PROPERTIES_BOOTSTRAP_SERVERS, bootstrapServers);
-
-        boolean wrapWithInlongMsg = format instanceof InLongMsgFormat;
-        Format realFormat = wrapWithInlongMsg ? ((InLongMsgFormat) format).getInnerFormat() : format;
-        if (realFormat instanceof JsonFormat
-                || realFormat instanceof AvroFormat
-                || realFormat instanceof CsvFormat) {
-            if (StringUtils.isEmpty(this.primaryKey)) {
-                options.put(KafkaConstant.CONNECTOR, KafkaConstant.KAFKA);
-                options.put(KafkaConstant.SCAN_STARTUP_MODE, kafkaScanStartupMode.getValue());
-                if (StringUtils.isNotEmpty(scanSpecificOffsets)) {
-                    options.put(KafkaConstant.SCAN_STARTUP_SPECIFIC_OFFSETS, scanSpecificOffsets);
-                }
-                if (StringUtils.isNotBlank(scanTimestampMillis)) {
-                    options.put(KafkaConstant.SCAN_STARTUP_TIMESTAMP_MILLIS, scanTimestampMillis);
-                }
-                options.putAll(delegateInlongFormat(realFormat.generateOptions(false), wrapWithInlongMsg));
-            } else {
-                options.put(KafkaConstant.CONNECTOR, KafkaConstant.UPSERT_KAFKA);
-                options.putAll(delegateInlongFormat(realFormat.generateOptions(true), wrapWithInlongMsg));
-            }
-        } else if (realFormat instanceof CanalJsonFormat
-                || realFormat instanceof DebeziumJsonFormat
-                || realFormat instanceof RawFormat) {
+        options.put(KafkaConstant.SCAN_STARTUP_MODE, kafkaScanStartupMode.getValue());
+        if (StringUtils.isEmpty(this.primaryKey)) {
             options.put(KafkaConstant.CONNECTOR, KafkaConstant.KAFKA);
-            options.put(KafkaConstant.SCAN_STARTUP_MODE, kafkaScanStartupMode.getValue());
-            if (StringUtils.isNotEmpty(scanSpecificOffsets)) {
-                options.put(KafkaConstant.SCAN_STARTUP_SPECIFIC_OFFSETS, scanSpecificOffsets);
-            }
-            if (StringUtils.isNotBlank(scanTimestampMillis)) {
-                options.put(KafkaConstant.SCAN_STARTUP_TIMESTAMP_MILLIS, scanTimestampMillis);
-            }
-            options.putAll(delegateInlongFormat(realFormat.generateOptions(false), wrapWithInlongMsg));
+            options.putAll(format.generateOptions(false));
         } else {
-            throw new IllegalArgumentException("kafka extract node format is IllegalArgument");
+            options.put(KafkaConstant.CONNECTOR, KafkaConstant.UPSERT_KAFKA);
+            options.putAll(format.generateOptions(true));
+        }
+        if (StringUtils.isNotEmpty(scanSpecificOffsets)) {
+            options.put(KafkaConstant.SCAN_STARTUP_SPECIFIC_OFFSETS, scanSpecificOffsets);
+        }
+        if (StringUtils.isNotBlank(scanTimestampMillis)) {
+            options.put(KafkaConstant.SCAN_STARTUP_TIMESTAMP_MILLIS, scanTimestampMillis);
         }
         if (StringUtils.isNotEmpty(groupId)) {
             options.put(KafkaConstant.PROPERTIES_GROUP_ID, groupId);
-        }
-        return options;
-    }
-
-    private Map<String, String> delegateInlongFormat(
-            Map<String, String> realOptions,
-            boolean wrapWithInlongMsg) {
-        if (!wrapWithInlongMsg) {
-            return realOptions;
-        }
-        Map<String, String> options = new HashMap<>();
-        for (Entry<String, String> entry : realOptions.entrySet()) {
-            String key = entry.getKey();
-            String value = entry.getValue();
-            if ("format".equals(key)) {
-                options.put("format", "inlong-msg");
-                options.put("inlong-msg.inner.format", value);
-            } else {
-                options.put("inlong-msg." + key, value);
-            }
         }
         return options;
     }

--- a/inlong-sort/sort-connectors/kafka/src/main/java/org/apache/inlong/sort/kafka/table/KafkaDynamicTableFactory.java
+++ b/inlong-sort/sort-connectors/kafka/src/main/java/org/apache/inlong/sort/kafka/table/KafkaDynamicTableFactory.java
@@ -326,10 +326,13 @@ public class KafkaDynamicTableFactory implements DynamicTableSourceFactory, Dyna
         final DecodingFormat<DeserializationSchema<RowData>> valueDecodingFormat =
                 getValueDecodingFormat(helper);
 
-        helper.validateExcept(PROPERTIES_PREFIX, DIRTY_PREFIX);
+        final String valueFormatPrefix = tableOptions.getOptional(FORMAT)
+                .orElse(tableOptions.get(VALUE_FORMAT));
 
+        // Validate the option data type.
+        helper.validateExcept(PROPERTIES_PREFIX, DIRTY_PREFIX, valueFormatPrefix);
+        // Validate the option values.
         validateTableSourceOptions(tableOptions);
-
         validatePKConstraints(
                 context.getObjectIdentifier(), context.getCatalogTable(), valueDecodingFormat);
 

